azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -1158,9 +1158,7 @@ func flattenAzureRmVirtualMachineScaleSetNetworkProfile(profile *compute.Virtual
 					config["load_balancer_inbound_nat_rules_ids"] = schema.NewSet(schema.HashString, inboundNatPools)
 				}
 
-				if properties.Primary != nil {
-					config["primary"] = *properties.Primary
-				}
+				config["primary"] = *properties.Primary
```

Comment:
can we revert this? the Azure API's return the state of the resource at the last time it was modified, meaning it's possible this won't be returned (and will crash here)